packages/event-handler/tests/unit/rest/middleware/compress.test.ts

```diff
@@ -0,0 +1,169 @@
+import { gzipSync } from 'node:zlib';
+import context from '@aws-lambda-powertools/testing-utils/context';
+import { Router } from 'src/rest/Router.js';
+import { beforeEach, describe, expect, it } from 'vitest';
+import { compress } from '../../../../src/rest/middleware/index.js';
+import { createSettingHeadersMiddleware, createTestEvent } from '../helpers.js';
+
+describe('Compress Middleware', () => {
```

Comment:
We should have assertions to check `isBase64Encoded` is set properly in these tests too.